pkg/provider/azure_vmssflex.go

```diff
@@ -444,13 +447,330 @@ func (fs *FlexScaleSet) GetNodeCIDRMasksByProviderID(providerID string) (int, in
 // EnsureHostInPool ensures the given VM's Primary NIC's Primary IP Configuration is
 // participating in the specified LoadBalancer Backend Pool, which returns (resourceGroup, vmasName, instanceID, vmssVM, error).
 func (fs *FlexScaleSet) EnsureHostInPool(service *v1.Service, nodeName types.NodeName, backendPoolID string, vmSetNameOfLB string) (string, string, string, *compute.VirtualMachineScaleSetVM, error) {
-	return "", "", "", nil, nil
+	serviceName := getServiceName(service)
+	name := mapNodeNameToVMName(nodeName)
+	vmssFlexName, err := fs.getNodeVmssFlexName(name)
+	if err != nil {
+		klog.Errorf("EnsureHostInPool: failed to get VMSS Flex Name %s: %v", name, err)
+		return "", "", "", nil, nil
+	}
+
+	// Check scale set name:
+	// - For basic SKU load balancer, return error as VMSS Flex does not support basic load balancer.
+	// - For single standard SKU load balancer, backend could belong to multiple VMSS, so we
+	//   don't check vmSet for it.
+	// - For multiple standard SKU load balancers, return nil if the node's scale set is mismatched with vmSetNameOfLB
+	needCheck := false
+	if !fs.useStandardLoadBalancer() {
+		return "", "", "", nil, fmt.Errorf("EnsureHostInPool: VMSS Flex does not support Basic Load Balancer")
```

Comment:
Will it be better if we report this error earlier? For example, once we detect vmssFlex+BLB, we report an error so no operations will be made. In this phase, the LB rules/probes/frontend has been updated.

Reply:
Do you mean to move this check to the top of this function or other places?

Reply:
Are the logics of  updating LB rules/probes/frontend executing in parallel? How does the place of checking BLB affect the executing of updating  LB rules/probes/frontend?